### DISCUSSION TITLE
flutter_supabase auth.signOut() should trigger 'onUnauthenticated'

### DISCUSSION THREAD

**Discussion by seenickcode** [supabase/supabase#3430] [Feature Requests]
It would be ideal to keep all auth state logic in a single place, such as [here](https://github.com/supabase/supabase/blob/master/examples/flutter-user-management/lib/components/auth_state.dart), but unfortunately, when calling `signOut`, nothing in this AuthState implementation gets triggered, thus making the user's code brittle, because we have to code logic [like thi](https://github.com/supabase/supabase/blob/master/examples/flutter-user-management/lib/screens/profile_screen.dart#L82) to update the route stack, vs having it all done in our  `AuthState` implementation.

**Comment by dshukertjr** [comment]
Hi @seenickcode !

That sounds like an amazing idea. If you don't mind, could you open an issue in `supabase_flutter`?

**Reply by seenickcode** [reply]
Done! https://github.com/supabase/supabase-flutter/issues/35
